contractcourt/chain_watcher.go

```diff
@@ -1452,3 +1433,101 @@ func (c *chainWatcher) handleCommitSpend(
 
 	return nil
 }
+
+// checkFundingSpend performs a non-blocking read on the spendNtfn channel to
+// check whether there's a commit spend already. Returns the spend details if
+// found.
+func (c *chainWatcher) checkFundingSpend() *chainntnfs.SpendDetail {
+	select {
+	// We've detected a spend of the channel onchain! Depending on the type
+	// of spend, we'll act accordingly, so we'll examine the spending
+	// transaction to determine what we should do.
+	//
+	// TODO(Roasbeef): need to be able to ensure this only triggers
+	// on confirmation, to ensure if multiple txns are broadcast, we
+	// act on the one that's timestamped
+	case spend, ok := <-c.fundingSpendNtfn.Spend:
+		// If the channel was closed, then this means that the notifier
+		// exited, so we will as well.
+		if !ok {
+			return nil
+		}
+
+		log.Debugf("Found spend details for funding output: %v",
+			spend.SpenderTxHash)
+
+		return spend
+
+	default:
+	}
+
+	return nil
+}
+
+// chanPointConfirmed checks whether the given channel point has confirmed.
+// This is used to ensure that the funding output has confirmed on chain before
+// we proceed with the rest of the close observer logic for taproot channels.
+func (c *chainWatcher) chanPointConfirmed() bool {
+	op := c.cfg.chanState.FundingOutpoint
+	confNtfn, err := c.cfg.notifier.RegisterConfirmationsNtfn(
```

Comment:
I think we need to be careful about some of these assumptions of insta conf sends here. This won't always be the case for the neutrino backend.

Reply:
cool we now moved the subscription earlier so it's more likely we can get this "on beat".

Reply:
good q - i think theoratically we could do a package of `[zero-conf funding tx -> coop close]`, but usually we want it to be confirmed to be safe.